src/components/ModelingMachineProvider.tsx

```diff
@@ -268,11 +268,25 @@ export const ModelingMachineProvider = ({
         ),
         'Set selection': assign(
           ({ context: { selectionRanges, sketchDetails }, event }) => {
-            const _event = event as Extract<
-              ModelingMachineEvent,
-              { type: 'Set selection' }
-            >
-            const setSelections = _event.data // this was needed for ts after adding 'Set selection' action to on done modal events
+            // this was needed for ts after adding 'Set selection' action to on done modal events
+            const setSelections =
+              ('data' in event &&
+                event.data &&
+                'selectionType' in event.data &&
+                event.data) ||
+              ('output' in event &&
+                event.output &&
+                'selectionType' in event.output &&
+                event.output) ||
+              null
```

Comment:
This is fine technically. My personal preference is to narrow down the event type by guarding against specific `event.type` properties but if we can receive multiple events here all having `data` or `output` properties, this is fine.